### PR TITLE
Fixing README for broken links

### LINKED ISSUE #4871: [BUG] Missing Links in Cosmos
**Describe the bug**
Some links in the README for Cosmos can't be found.

***Exception***
Example: in Cosmos under the README (https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/cosmos), the link to the HelloWorldExample.java doesn't work
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/examples/src/main/java/com/azure/data/cosmos/examples/HelloWorldExample.java

***Snippet***
Other link
Cosmos README:
-  Examples - https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/examples/src/test/java/com/microsoft/azure/cosmosdb/rx/examples
- Can't seem to find the API documentation: https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/cosmos#api-documentation as well as the Usage Code Sample 
- Guide for Prod doesn't jump to Guide for Production

### PR DESCRIPTION
**Pull request by kirankumarkolli**
Closes #4871

### CODE REVIEW

File: sdk/cosmos/README.md [hunk 1]
```diff
@@ -1,27 +1,72 @@
...
+- (Optional) SLF4J is a logging facade.
+- (Optional) [SLF4J binding](http://www.slf4j.org/manual.html) is used to associate a specific logging framework with SLF4J.
+- (Optional) Maven
+
+SLF4J is only needed if you plan to use logging, please also download an SLF4J binding which will link the SLF4J API with the logging implementation of your choice. See the [SLF4J user manual](http://www.slf4j.org/manual.html) for more information.
+
+<!-- TODO - update once JavaDoc is published
+## API Documentation
+
+Javadoc is available [here](https://azure.github.io/azure-cosmosdb-java/2.4.0/com/microsoft/azure/cosmosdb/rx/AsyncDocumentClient.html).
```
Comment (moderakh) [inline sdk/cosmos/README.md]: This refers to javadoc of async v2.

Comment (kirankumarkolli) [reply sdk/cosmos/README.md]: @kushagraThapar what's the reference link to use here?

Comment (kushagraThapar) [reply sdk/cosmos/README.md]: @kirankumarkolli It should be this: https://azuresdkdocs.blob.core.windows.net/$web/java/azure-cosmos/4.0.0-preview.1/index.html
Once we GA, we can update this link to GA version.

**Comment by conniey** [comment]
Closed due to inactivity and merge conflicts now.

**Pull request closed without merging**